src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -345,27 +345,18 @@ private static uint HandleEventPeerCertificateReceived(State state, ref Connecti
                     {
                         unsafe
                         {
-                            ReadOnlySpan<QuicBuffer> quicBuffer;
+                            ReadOnlySpan<QuicBuffer> quicBuffer = new ReadOnlySpan<QuicBuffer>((void*)connectionEvent.Data.PeerCertificateReceived.PlatformCertificateHandle, sizeof(QuicBuffer));
+                            certificate = new X509Certificate2(new ReadOnlySpan<byte>(quicBuffer[0].Buffer, (int)quicBuffer[0].Length));
```

Comment:
We're guaranteed quicBuffer.Length > 0?

Reply:
Nevermind, I missed the `sizeof(QuicBuffer)` as the length above.

Reply:
BTW the relevant code is here:
https://github.com/microsoft/msquic/blob/5f9b5482f0fcabb970033cf72dfcde2f472f0262/src/platform/tls_openssl.c#L218-L228

It does not check for length but since certificate exist and i2d_X509 did not fail I assume the length is positive. 
Similar for the chain aka pkcs7 blob.